File: 04-counting-elements/perm-check/c/test_perm_check.cpp
#include <iostream>
#include "quick_test.h"
#include "gtest/gtest.h"
#include "perm_check.h"

class PermCheckTest : public QuickTest {};

TEST(PermCheckTest, Test_1) {
  int A[] = {4, 1, 3, 2};
  int N = 4;

  int result = solution(A, N);

  EXPECT_EQ(result, 1);
}

TEST(PermCheckTest, Test_2) {
  int A[] = {4, 1, 3};
  int N = 2;

  int result = solution(A, N);

  EXPECT_EQ(result, 0);
}

int main(int argc, char **argv) {
  ::testing::InitGoogleTest(&argc, argv);

  int ret = RUN_ALL_TESTS();

  std::cout << "Press Enter to exit";
  std::cin.get();

  return ret;
}

File: 04-counting-elements/perm-check/python/test_perm_check.py
import pytest

from perm_check import solution

@pytest.mark.parametrize("test_input, expected", [([4, 1, 3, 2], 1), ([4, 1, 3], 0), ([1, 1], 0), ([4, 1, 3, 2, 1], 0)])
def test_solution(test_input, expected):
    result = solution(test_input)
    assert result == expected
